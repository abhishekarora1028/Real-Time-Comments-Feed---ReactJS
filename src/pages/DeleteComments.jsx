import React, {useEffect, useState} from 'react';
import { CommentService } from '../services/CommentService';
import { Container, Row, Col, Button} from 'react-bootstrap';
import { Link } from 'react-router-dom';
import socketIOClient from "socket.io-client";
import './Layout.scss';
const socket = socketIOClient('http://localhost:3001');

const DeleteComments = (props) => {

const [comments, setComments] = useState([]);
 
useEffect(() => {
  CommentService.get('/getComments').then(
    result => {
      setComments(result)
    });
}, []);
      
const hasComments = (comments) => {
      return (
          <div className="deletecomments">
            <h3> Click this button to delete all comments </h3>
            <Button className="btn btn-danger" onClick={deleteComments}> Delete {comments.length} Comment{comments.length > 1 && 's'}</Button>
            <p className="text-right">
            <Link to="/">
                No, Go Back Home
              </Link>
            </p>
          </div>
      );
    }

const deleteComments = () => {
      if (window.confirm('Are you sure you want to delete all the comments?')) {
        CommentService.delete('/deleteComments').then(() => {
          socket.emit("initial_data");
          setComments([]);
        });
      }
    }

 return (
      <Container className="p-3 page" fluid="sm">
        <Row>
          <Col className="text-center">
           {comments.length ? hasComments(comments) : 'Sorry, no comments found'}
          </Col>
        </Row>
      </Container>
      );
    }
 
  export default DeleteComments;