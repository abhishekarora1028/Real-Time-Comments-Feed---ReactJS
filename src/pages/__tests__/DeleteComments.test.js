import React from 'React';
import { render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import DeleteComments from '../DeleteComments';
import socketIOClient from "socket.io-client";

afterEach(cleanup);

describe('DeleteComments', () => {

    it('Should render the Delete Comments Page', () =>{
        const { asFragment } = render(<DeleteComments/>);
        expect(asFragment(<DeleteComments/>)).toMatchSnapshot();
    });
});