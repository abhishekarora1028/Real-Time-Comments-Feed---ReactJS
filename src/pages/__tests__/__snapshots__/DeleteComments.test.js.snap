// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteComments Should render the Delete Comments Page 1`] = `
<DocumentFragment>
  <div
    class="p-3 page container-sm"
  >
    <div
      class="row"
    >
      <div
        class="text-center col"
      >
        Sorry, no comments found
      </div>
    </div>
  </div>
</DocumentFragment>
`;
